ember-file-upload/src/system/upload.ts

```diff
@@ -52,6 +52,28 @@ function normalizeOptions(
   return options;
 }
 
+function updateRate(file: UploadFile) {
+  const updatedTime = new Date().getTime();
+
+  // If there's a previous recording, we can calculate a rate from that
+  if (file.timestampWhenProgressLastUpdated) {
+    const timeElapsedSinceLastUpdate =
+      updatedTime - file.timestampWhenProgressLastUpdated;
+
+    const bytesTransferredSinceLastUpdate =
+      file.loaded - file.bytesWhenProgressLastUpdated;
+
+    // Divide by elapsed time to get bytes per millisecond
+    const rate = bytesTransferredSinceLastUpdate / timeElapsedSinceLastUpdate;
+
+    file.rates = [...file.rates, rate];
```

Comment:
Would it be more performant to use a `TrackedArray`?

Reply:
I'm not sure. I say leave it unless we have problems.